aten/src/ATen/core/op_registration/op_registration_test.cpp

```diff
@@ -2157,6 +2157,24 @@ TEST(OperatorRegistrationTest, getRegistrationsForDispatchKey) {
   ASSERT_TRUE(std::includes(all_ops.begin(), all_ops.end(), cpu_ops.begin(), cpu_ops.end(), cmp_lambda));
 }
 
+Tensor symint_op(Tensor self, c10::SymInt length) {
+  return self;
+}
+
+TEST(OperatorRegistrationTest, TestSymNonSymCompatibility) {
+  auto m = MAKE_TORCH_LIBRARY(_test);
+  m.def("_test::symint_op(Tensor self, SymInt length) -> Tensor");
+  auto m_cpu = MAKE_TORCH_LIBRARY_IMPL(_test, CPU);
+  m_cpu.impl("symint_op", c10::DispatchKey::CPU, TORCH_FN(symint_op));
+
+  auto opHandle = c10::Dispatcher::singleton().findSchemaOrThrow(
+      "_test::symint_op", "");
+
+  expectThrows<c10::Error>([&] {
+    opHandle.typed<Tensor(const Tensor&, c10::SymInt)>().call(dummyTensor(c10::DispatchKey::CPU), c10::SymInt(4));
+  }, "Tried to access or call an operator with a wrong signature");
+}
```

Comment:
This test just checks Tensor vs const Tensor& mismatch, right? Could we add one for SymInt as well?